### PR TITLE
CORE-145: dependency updates

### PR DESCRIPTION
**Pull request by slr71**
The primary purpose of this change was to switch to `compojure-api` version 1.1.12, which supports the `charset` attribute of the `Boundary` header. I thought I'd update several other dependencies at the same time.

### CODE REVIEW

File: project.clj
Comment (psarando) [inline project.clj:12]: Should we cut a `3.0.3` release for `clojure-commons` and use that here?

Comment (slr71) [reply project.clj:12]: That sounds good to me. I figured I'd cut releases and update dependencies after merging these PRs.

Comment (psarando) [reply project.clj:12]: That works too 👌

**Review by psarando — APPROVED** [review]
LGTM :shipit: when ready.

**Pull request merged**
